cnd/src/main.rs

```diff
@@ -142,10 +142,15 @@ fn main() -> anyhow::Result<()> {
     };
 
     runtime.block_on(load_swaps::load_swaps_from_database(deps.clone()))?;
-    runtime.spawn(spawn_warp_instance(settings, deps));
 
-    // Block the current thread.
-    ::std::thread::park();
+    let listen_addr = settings.http_api.socket;
+    // This is racey but gives us some protection, if the race condition is hit it
+    // just means cnd will start even though the HTTP server did not come up.
+    if !port_is_available(listen_addr.port()) {
+        return Err(anyhow::anyhow!("HTTP port is unavailable: {}", listen_addr));
+    }
```

Comment:
```suggestion
```

I don't think this is needed, the panic from warp is clear enough: `thread 'main' panicked at 'error binding to 0.0.0.0:8000: error creating server listener: Address already in use (os error 48)', <::std::macros::panic macros>:5:6
`

Reply:
That is the whole point of this ticket, the panic from warp only panics the warp thread not the main thread so cnd continues to run.

Reply:
If you use `block_on` then the main thread panics too.